src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,86 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppSelector } from '../../hooks';
+import { RootState } from '../../app/store';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+interface TodoModalProps {
+  onClose: (isOpen: boolean) => void;
+}
 
-      <Loader />
+export const TodoModal: React.FC<TodoModalProps> = ({ onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const todo = useAppSelector((state: RootState) => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  if (!todo) {
+    return null;
+  }
+
+  const { title, id, completed, userId } = todo;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useEffect(() => {
+    const fetchUser = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedUser = await getUser(userId);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+        setTimeout(() => {
+          setUser(fetchedUser);
+          setIsLoading(false);
+        }, 300);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Ошибка при загрузке данных пользователя:', error);
+        setIsLoading(false);
+      }
+    };
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    fetchUser();
+  }, [userId]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {id ? `Todo #${id}` : 'Todo'}
+            </div>
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => onClose(false)}
+            />
```

Comment:
This is a code quality issue. You should avoid using arrow functions in props if possible. Every time this component re-renders, a new function is created. This can lead to unnecessary re-renders and performance issues especially if this component is rendered often or has many instances. Consider moving this function to the component body and reference it in the onClick prop.